docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,216 @@
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+5. Data formatting and handling by the event streaming service is out of the scope of this epic. The event streaming service should be able to process the data from all versions of the on-premises environment.
+
+
+## Proposal
+
+### Data Sending Flow
+
+All of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Data will be sent when one of the following happens:
+1. A cluster installation attempt finishes (regardless of success or failure)
+2. A shutdown trigger: cluster is getting deleted, all hosts are removed from a cluster, or the API server is getting shut down
+3. An interval of time has passed and the cluster has not changed state (meaning the installation is abandoned/forgotten), just send the data we have until that point
+
+Before the data is sent, Assisted Service will get a remote configuration from an endpoint we specify.
+
+Sending data should not interfere with customer workloads and deployment
+
+### Assisted-Service Changes 
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query and send data (data sending flow event 1 above)
+- Needs code changes when a cluster is deleted, all hosts are removed from a cluster, and the API server is shutting down to query and send data (data sending flow event 2 above)
+- Needs code changes to periodically check if a cluster state hasn't changed in a given amount of time to query and send data (data sending flow event 3 above)
+- Needs to know how to format the data to send to the data hosting service
+    - Includes sanitizing the data such as hashing user's emails like in the [event scraper](https://github.com/openshift-assisted/assisted-events-scrape/blob/master/assisted-events-scrape/utils/anonymizer.py#L25)
+- Needs to send data using best effort and with retry logic 
+- Needs to expose an option for users to opt out of this feature
+- Needs to know where and how to get the remote configuration 
+
+### Additional Changes
+
+- Documentation regarding terms and conditions?
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment
+- Host a remote configuration (using a CDN)
+
+### Remote Configuration
+
+The remote configuration will specify configuration data for the assisted-service such as `DATA_COLLECTION_URL`. This can be cached in a CDN.
+
+The assisted service will get this remote configuration:
+
+1. When the assisted service initially starts
+2. Every time the assisted service will send data it'll get this configuration first
```

Comment:
why? how often it will be changed?

Reply:
IIUC I don't think this is necessary.
Mostly because I don't see this URL changing any more frequently than whatever mechanism you are planning to use to retrieve this URL.

Reply:
Thanks for the perspective! Modified the doc to move remote configuration to a potential future enhancement.